src/errors.rs

```diff
@@ -1,10 +1,20 @@
 use std::error;
+use std::fmt::Display;
 use std::io::Error as IoError;
 
 use thiserror::Error;
 
 use crate::codec::Error as CodecError;
 
+#[derive(Debug, Error)]
+pub struct Corruption(pub Box<dyn error::Error + Send + Sync>);
```

Comment:
Why is this extra abstraction needed?

Reply:
There's already `Other(#[from] Box<dyn error::Error + Send + Sync>)` in `Error`. Without another abstraction the errors created by `box_err!` cannot decide which item to convert into.

Reply:
Then let's replace box_err with `String`